node/src/bin/manager.rs

```diff
@@ -1585,3 +1560,222 @@ async fn main() -> anyhow::Result<()> {
+fn make_graphman_context(ctx: Context) -> GraphmanContext {
+    let logger = ctx.logger.clone();
+    let notification_sender = ctx.notification_sender();
+    let (store, pool) = ctx.store_and_primary();
+
+    GraphmanContext {
+        pool,
+        notification_sender,
+        store,
+        logger,
+    }
+}
+
+async fn execute_deployment_info(
+    ctx: Context,
+    deployment: Option<DeploymentSearch>,
+    current: bool,
+    pending: bool,
+    status: bool,
+    used: bool,
+    all: bool,
+) -> anyhow::Result<()> {
+    use graphman::deployment_search::DeploymentFilters;
+    use graphman::deployment_search::DeploymentSelector;
+
+    let deployment = match deployment {
+        Some(deployment) => make_deployment_selector(deployment),
+        None if all => DeploymentSelector::All,
+        None => bail!("Please specify a deployment or use --all to list all deployments"),
+    };
+
+    let filters = DeploymentFilters {
+        included_versions: Some(make_deployment_version_filter(current, pending, used)),
+    };
+
+    let status = if !status {
+        DeploymentStatusSelector::Skip
+    } else {
+        DeploymentStatusSelector::Load
+    };
+
+    let command = DeploymentInfoCommand {
+        deployment,
+        filters,
+        status,
+    };
+
+    let ctx = make_graphman_context(ctx);
+    let deployments = command.execute(ctx).await?;
+
+    print_deployments(deployments);
+
+    Ok(())
+}
+
+async fn execute_pause_deployment(
+    ctx: Context,
+    deployment: DeploymentSearch,
+) -> anyhow::Result<()> {
+    let deployment = make_deployment_selector(deployment);
+    let command = PauseDeploymentCommand { deployment };
+    let ctx = make_graphman_context(ctx);
+
+    let _paused = command.execute(ctx).await?;
+
+    println!("Operation completed successfully");
+    Ok(())
+}
+
+async fn execute_resume_deployment(
+    ctx: Context,
+    deployment: DeploymentSearch,
+) -> anyhow::Result<()> {
+    let deployment = make_deployment_selector(deployment);
+    let command = ResumeDeploymentCommand { deployment };
+    let ctx = make_graphman_context(ctx);
+
+    let _resumed = command.execute(ctx).await?;
+
+    println!("Operation completed successfully");
+    Ok(())
+}
+
+async fn execute_restart_deployment(
+    ctx: Context,
+    deployment: DeploymentSearch,
+    delay: Duration,
+) -> anyhow::Result<()> {
+    let deployment = make_deployment_selector(deployment);
+    let command = RestartDeploymentCommand { deployment, delay };
+    let ctx = make_graphman_context(ctx);
+    let store = Arc::new(GraphmanStore::new(ctx.pool.clone()));
+    let ctx = GraphmanExtensionContext::new(ctx);
+
+    let _restarted = command
+        .layer(IdentifyCommand::new(CommandKind::RestartDeployment.into()))
+        .layer(HandleBrokenExecutions::new(store.clone()))
+        .layer(PreventDuplicateExecutions::new(store.clone()))
+        .layer(TrackExecution::new(store))
+        .execute(ctx)
+        .await?;
```

Comment:
For the CLI, that's not really what you want: you want the execution to be synchronous so that execution stops immediately when the user hits ^C.